.github/workflows/end2endtest.yml

```diff
@@ -0,0 +1,34 @@
+name: End2EndTests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 4
+      matrix:
+        os: [ubuntu-latest, macos-latest]
```

Comment:
IIRC we found that git-theta should work fine on windows as long as it wasn't using FAT32 and trying to track giant files, right? Any reason not to include windows-latest here?

Reply:
I tried to add windows to the unittest a while ago and installing pytorch took so long that it kept timing out.

Also in this case the test scripts are all written in bash so I don't know if that would work on windows (we would need to use the linux subsystem somehow?)

Reply:
I might be wrong but I think if we specify `shell: bash` (and resolve the pytorch install timeout) we can use windows-latest. bash is supported by all platforms; windows-latest uses powershell by default but that can be overriden. https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell We can deal with this in a follow-up though.